src/test/api/contract-scenario-headers-test.js

```diff
@@ -132,27 +130,31 @@ function fullMatchRequest() {
         .set('stringRequired', 'a string')
         .set('numberRequired', '234')
         .set('stringOptional', 'also a string')
-        .set('value', 'some value');
+        .set('value', 'some value')
+        .set('jsonValue', JSON.stringify({key3: 'val3', key1: 'val1'}));
 }
 
 function onlyRequiredRequest() {
     return request.get('/header-types/only-required-match')
         .set('stringRequired', 'a string')
         .set('numberRequired', '234')
-        .set('value', 'some value');
+        .set('value', 'some value')
+        .set('jsonValue', JSON.stringify({key3: 'val3', key1: 'val1'}));
 }
 
 function missingRequiredRequest() {
     return request.get('/header-types/missing-required')
         .set('numberRequired', '234')
-        .set('value', 'some value');
+        .set('value', 'some value')
+        .set('jsonValue', JSON.stringify({key3: 'val3', key1: 'val1'}));
 }
 
 function wrongTypeRequest() {
     return request.get('/header-types/wrong-type')
         .set('stringRequired', 'a string')
         .set('numberRequired', '234')
-        .set('value', 'some value');
+        .set('value', 'some value')
+        .set('jsonValue', JSON.stringify({key3: 'val3', key1: 'val1'}));
 }
 
```

Comment:
Should have an api test or two here that are explicitly showing failing requests for mismatching json, as well as matching json with out of order keys. I find these also work very well as examples.